### PR TITLE
Add chi_p, chi_a to conversions.py

### PR DESCRIPTION
**Pull request by cmbiwer**
Adds conversions to/from ``chi_p``, ``chi_a``, and angles from: https://dcc.ligo.org/LIGO-T1700056

### CODE REVIEW

File: pycbc/conversions.py [hunk 1]
```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
     ss[mask] = spin1[mask]
     return _formatreturn(ss)
 
+def primary_xi(spinx, spiny):
+    """Returns the effective precession spin argument for the larger mass.
+    """
+    return chi_perp_from_spin1x_spin1y(spinx, spiny)
```
Comment (cdcapano) [inline pycbc/conversions.py]: Make this take mass1, mass2, spin1x, spin1y, spin2x, and spin2y. I know you only need the spins for xi1, but the `primary` label indicates that the returned values are guaranteed to be for the larger object. If you make it take mass1, mass2, you can guarantee that that is true for an array of values in which mass1 is not always greater than mass2 (as might be the case in a PE run).

That said, it is nice having a simple function that return xi1 given two spins. So, rename this function `xi1_from_spinx_spiny`, and make it explicit in the doc string that it is assumed the spins are from the larger body. Then add a function `primary_xi` that takes in the masses and spins of both objects, uses the `primay_spin` function to pull out the spinx and y of the larger body, then passes it to `xi1_from_spinx_spiny`.

File: pycbc/conversions.py [hunk 2]
```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
...
+
+def secondary_xi(mass1, mass2, spinx, spiny):
+    """Returns the effective precession spin argument for the smaller mass.
+    """
+    q = q_from_mass1_mass2(mass1, mass2)
+    a1 = 2 + 3 * q / 2
+    a2 = 2 + 3 / (2 * q)
+    return a1 / (q**2 * a2) * chi_perp_from_spin1x_spin1y(spinx, spiny)
```
Comment (cdcapano) [inline pycbc/conversions.py]: Same comment as above... rename this `xi2_from_mass1_mass2_spinx_spiny`, and add a new function called `secondary_xi` that takes mass1, mass2, spin1x, spin1y, spin2x, spin2y, pulls out the secondary spins, and passes that to the former function.

File: pycbc/conversions.py [hunk 3]
```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
...
+
+def chi_perp_from_spin1x_spin1y(spin1x, spin1y):
+    """Returns the in-plane spin from spin1x and spin1y.
+    """
+    return numpy.sqrt(spin1x**2 + spin1y**2)
+
+def chi_perp_from_spin2x_spin2y(spin2x, spin2y):
+    """Returns the in-plane spin from spin2x and spin2y.
+    """
+    return chi_perp_from_spin1x_spin1y(spin2x, spin2y)
```
Comment (cdcapano) [inline pycbc/conversions.py]: These two functions are redundant. Rename `chi_perp_from_spin1x_spin1y`, `chi_perp_from_spinx_spiny`, and get rid of the second function.

File: pycbc/conversions.py [hunk 4]
```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
...
+
+def phi1_from_spin1x_spin1y(spin1x, spin1y):
+    """Returns phi1 for primary mass.
+    """
+    return numpy.arctan(spin1y / spin1x)
+
+def phi2_from_spin2x_spin2y(spin2x, spin2y):
+    """Returns phi2 for secondary mass.
+    """
+    return phi1_from_spin1x_spin1y(spin2x, spin2y)
```
Comment (cdcapano) [inline pycbc/conversions.py]: Same comment as above... just name this `phi_from_spinx_spiny` and get rid of the second function.

File: pycbc/conversions.py [hunk 5]
```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
...
+    return (mass1 + mass2) / (2 * mass1) * (chi_eff - chi_a)
+
+def spin2z_from_mass1_mass2_chi_eff_chi_a(mass1, mass2, chi_eff, chi_a):
+    """Returns spin2z.
+    """
+    return (mass1 + mass2) / (2 * mass2) * (chi_eff + chi_a)
+
+def spin1x_from_xi1_phi_a_phi_s(xi1, phi_a, phi_s):
+    """Returns spin2x for primary mass.
+    """
```
Comment (cdcapano) [inline pycbc/conversions.py]: `spin2x for` -> `x-component of the` in the doc string.

File: pycbc/conversions.py [hunk 6]
```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
...
+    phi1 = phi1_from_phi_a_phi_s(phi_a, phi_s)
+    return xi1 * numpy.cos(phi1)
+
+def spin1y_from_xi1_phi_a_phi_s(xi1, phi_a, phi_s):
+    """Returns spin2y for primary mass.
+    """
```
Comment (cdcapano) [inline pycbc/conversions.py]: `spin2y for` -> `y-component of the` in the doc string.

File: pycbc/conversions.py [hunk 7]
```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
...
+    phi1 = phi1_from_phi_a_phi_s(phi_s, phi_a)
+    return xi1 * numpy.sin(phi1)
+
+def spin2x_from_mass1_mass2_xi2_phi_a_phi_s(mass1, mass2, xi2, phi_a, phi_s):
+    """Returns spin2x for secondary mass.
+    """
```
Comment (cdcapano) [inline pycbc/conversions.py]: `spin2x for` -> `x-component of the`

File: pycbc/conversions.py [hunk 8]
```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
...
+    chi_perp = chi_perp_from_mass1_mass2_xi2(mass1, mass2, xi2)
+    phi2 = phi2_from_phi_a_phi_s(phi_a, phi_s)
+    return chi_perp * numpy.cos(phi2)
+
+def spin2y_from_mass1_mass2_xi2_phi_a_phi_s(mass1, mass2, xi2, phi_a, phi_s):
+    """Returns spin2y for secondary mass.
+    """
```
Comment (cdcapano) [inline pycbc/conversions.py]: `spin2y for` -> `y-component of the`

**Comment by cmbiwer** [comment]
I've made the changes you've asked for.

**Comment by cmbiwer** [comment]
I've made those changes.

**Pull request merged**